corehq/messaging/scheduling/forms.py

```diff
@@ -1879,14 +1879,15 @@ def get_advanced_survey_layout_fields(self):
     @cached_property
     def language_list(self):
         sms_translations = get_or_create_sms_translations(self.domain)
-        result = set(sms_translations.langs)
+        result = sms_translations.langs     # maintain order set on languages config page
 
+        # add any languages present in alert but deleted from languages config page
         if self.initial_schedule:
-            result |= self.initial_schedule.memoized_language_set
+            initial_langs = self.initial_schedule.memoized_language_set
+            initial_langs.discard('*')
+            result += list(self.initial_schedule.memoized_language_set - set(result))
```

Comment:
It looks like ordering will be lost on this line. Is that not an issue?

Reply:
That's okay. The languages that get added here are those that are used in the conditional alert but don't appear on the SMS Languages page, presumably because they were deleted from the project. Since they aren't on the language config page, there isn't a correct ordering for them.